source/simulator/nullspace.cc

```diff
@@ -221,13 +221,6 @@ namespace aspect
 
           }
 
-#ifdef DEBUG
-      Assert(n_left_to_find == 0, ExcMessage("Error, couldn't find a velocity DoF to constrain."));
-
-      for (unsigned int d=0; d<dim; ++d)
-        Assert(vel_idx[d] != numbers::invalid_dof_index,
-               ExcMessage("Error, couldn't find a velocity DoF to constrain."));
-#endif
```

Comment:
The old logic is the following:
1. each processor find a valid dof index
2. we take the minimum over the index and constrain that DoF.
The assert checks that each processor is successful in step 1, which is impossible if you don't own any DoFs. So this check needs to go.

If you look at the new Assert below we catch the case if no processor is able to find a dof to constrain.

So yes, this should work correctly and generate wrong results.